app/src/pages/inside/launchesPage/launchSuiteGrid/defectStatistics/donutChart/donutChart.jsx

```diff
@@ -0,0 +1,96 @@
+import { Component } from 'react';
+import classNames from 'classnames/bind';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+import { DefectTypeTooltip } from 'pages/inside/launchesPage/defectTypeTooltip';
+import { withHoverableTooltip } from 'components/main/tooltips/hoverableTooltip';
+import { projectConfigSelector, defectColorsSelector } from 'controllers/project';
+import styles from './donutChart.scss';
+
+const cx = classNames.bind(styles);
+@withHoverableTooltip({
+  TooltipComponent: DefectTypeTooltip,
+  data: {
+    width: 235,
+    align: 'right',
+    noArrow: true,
+    desktopOnly: true,
+    verticalOffset: -10,
+  },
+})
+@connect((state) => ({
+  projectConfig: projectConfigSelector(state),
+  defectColors: defectColorsSelector(state),
+}))
+export class DonutChart extends Component {
+  static propTypes = {
+    type: PropTypes.string,
+    data: PropTypes.object.isRequired,
+    diameter: PropTypes.number.isRequired,
+    strokeWidth: PropTypes.number.isRequired,
+    projectConfig: PropTypes.object.isRequired,
+    defectColors: PropTypes.object.isRequired,
+  };
+  static defaultProps = {
+    type: '',
+  };
+  componentWillMount() {
+    const defects = this.props.data;
+    let offset = 75;
+    Object.keys(this.props.data).forEach((defect) => {
+      if (defect !== 'total') {
+        const val = defects[defect];
+        const percents = val / defects.total * 100;
+
+        this.chartData.push({
+          id: defect,
+          value: percents,
+          color: this.props.defectColors[defect],
+          offset: 100 - offset,
+        });
+        offset += percents;
+      }
+    });
+  }
+  chartData = [];
```

Comment:
I think it's better to move it to the the state and initialize this state in the constructor instead of `componentWillMount` (which will be deprecated in the future).